docs/ossm/ossm2-migration/cluster-wide/README.md

```diff
@@ -0,0 +1,365 @@
+# OpenShift Service Mesh 2.6 Cluster wide --> 3 Migration guide
+This guide is for users who are currently running `ClusterWide` OpenShift Service Mesh 2.6 migrating to OpenShift Service Mesh 3.0. You should first read [this document comparing OpenShift Service Mesh 2 vs. OpenShift Service Mesh 3](../../ossm2-vs-ossm3.md) to familiarize yourself with the concepts between the two versions and the differences in managing the workloads and addons.
+
+## Migrating OpenShift Service Mesh 2.6 Cluster wide to OpenShift Service Mesh 3
+
+### Prerequisites
+- you have read [OpenShift Service Mesh 2 vs. OpenShift Service Mesh 3](../../ossm2-vs-ossm3.md)
+- you have completed all the steps from the [pre-migration checklist](../README.md#pre-migration-checklist)
+- you have verified that your 2.6 `ServiceMeshControlPlane` is using `ClusterWide` mode
+- Red Hat OpenShift Service Mesh 3 operator is installed
+- `IstioCNI` is installed
+- `istioctl` is installed
+
+### Procedure
+In this example, we'll be using the [bookinfo demo](https://raw.githubusercontent.com/Maistra/istio/maistra-2.6/samples/bookinfo/platform/kube/bookinfo.yaml) but you can follow these same steps with your own workloads.
+
+#### Plan the migration
+There will be two cluster wide Istio control planes running during the migration process so it's necessary to plan the migration steps in advance in order to avoid possible conflicts between the two control planes.
+
+There are a few conditions which must be verified to ensure a successful migration:
+- both control planes must share the same root certificate
+
+  This can be achieved by installing the 3.0 control plane to the same namespace as 2.6 control plane. The migration procedure below shows how to verify the root cert is shared.
+- 3.0 control plane must have access to all namespaces in 2.6 mesh
+
+  During the migration, some proxies will be controlled by the 3.0 control plane while others will still be controlled by the 2.6 control plane. To assure the communication still works, both control planes must be aware of the same set of services. You must verify that:
+  1. there are no Network Policies blocking the traffic
+
+     OpenShift Service Mesh 3.0 is no longer managing Network Policies and it's up to users to assure that existing Network Policies are not blocking the traffic for OpenShift Service Mesh 3.0 components. See Migration of Network Policies documentation for details.
```

Comment:
"No network policies blocking traffic" makes sense, but I am a not clear on the next sentence. Are we saying users should migrate network policies to OSSM 3 before doing the migration? Could that cause problems migrating traffic?

The pre-condition of "no network policies to block traffic" during the migration seems like it may be a separate issue from "I want network policies in OSSM 3 that does the same thing that the OSSM 2 generated network policies did". The pre-migration checklist already covered that network policy management should be disabled...if the user is going to re-introduce network policies for OSSM 3, should that be done before the migration or afterwards?

Reply:
Yes, before the migration the 2.6 feature for NP management must be disabled. Users are free to decide if they will re-create the NP manually or not. It's also up to them if it's done before or after the migration. The important part is that newly created NP must not block the traffic for either of the control planes if done before. If done after, users only care about 3.0 control plane. I tried to expand this part.